packages/cli/src/services/role.service.ts

```diff
@@ -132,4 +144,68 @@ export class RoleService {
 			return acc;
 		}, new Set());
 	}
+
+	addScopes(
+		rawWorkflow: ListQuery.Workflow.WithSharing | ListQuery.Workflow.WithOwnedByAndSharedWith,
+		user: User,
+		userProjectRelations: ProjectRelation[],
+	): ListQuery.Workflow.WithScopes;
+	addScopes(
+		rawCredential:
+			| ListQuery.Credentials.WithSharing
+			| ListQuery.Credentials.WithOwnedByAndSharedWith,
+		user: User,
+		userProjectRelations: ProjectRelation[],
+	): ListQuery.Credentials.WithScopes;
+	addScopes(
+		rawEntity:
+			| ListQuery.Workflow.WithSharing
+			| ListQuery.Credentials.WithOwnedByAndSharedWith
+			| ListQuery.Credentials.WithSharing
+			| ListQuery.Workflow.WithOwnedByAndSharedWith,
+		user: User,
+		userProjectRelations: ProjectRelation[],
+	): ListQuery.Workflow.WithScopes | ListQuery.Credentials.WithScopes {
+		const shared = rawEntity.shared;
+		const entity = rawEntity as ListQuery.Workflow.WithScopes | ListQuery.Credentials.WithScopes;
+
+		Object.assign(entity, {
+			scopes: [],
+		});
+
+		if (shared === undefined) {
+			return entity;
+		}
+
+		if (!('active' in entity) && !('type' in entity)) {
+			throw new ApplicationError('Cannot detect if entity is a workflow or credential.');
+		}
+
+		const globalScopes = this.getRoleScopes(user.role).filter((s) =>
+			s.startsWith('active' in entity ? 'workflow:' : 'credential:'),
+		);
```

Comment:
I've noticed a difference here, when adding scopes to workflows and credentials, we check only for scopes that `startsWith` but when fetching for projects, we just add all global scopes. Should we perhaps do the same for projects?

Reply:
I just tried changing the code to accomodate this comment and noticed that the project-specific scopes contain `workflow:*` and `credential:*` scopes, so they span over more than just `project:*` hence it makes senes to return everything here. I wonder whether we can optimiza this in the future, but it should be fine for now.